frame/contracts/src/wasm/runtime.rs

```diff
@@ -658,21 +672,26 @@ where
 			self.charge_gas(costs)?;
 		}
 
-		self.memory
-			.set(out_ptr, buf)
-			.and_then(|_| self.memory.set(out_len_ptr, &buf_len.encode()))
-			.map_err(|_| Error::<E::T>::OutOfBounds)?;
-
-		Ok(())
+		self.write_sandbox_memory(memory, out_ptr, buf)?;
+		self.write_sandbox_memory(memory, out_len_ptr, &buf_len.encode())
 	}
 
 	/// Write the given buffer to the designated location in the sandbox memory.
 	///
 	/// Returns `Err` if one of the following conditions occurs:
 	///
 	/// - designated area is not within the bounds of the sandbox memory.
-	fn write_sandbox_memory(&mut self, ptr: u32, buf: &[u8]) -> Result<(), DispatchError> {
-		self.memory.set(ptr, buf).map_err(|_| Error::<E::T>::OutOfBounds.into())
+	fn write_sandbox_memory(
+		&mut self,
```

Comment:
```suggestion
		&self,
```
seems like `self` ref doesn't need to be mutable here anymore